source/fuzz/fuzzer_pass_replace_adds_subs_muls_with_carrying_extended.cpp

```diff
@@ -37,40 +37,39 @@ FuzzerPassReplaceAddsSubsMulsWithCarryingExtended::
     ~FuzzerPassReplaceAddsSubsMulsWithCarryingExtended() = default;
 
 void FuzzerPassReplaceAddsSubsMulsWithCarryingExtended::Apply() {
-  for (auto& function : *GetIRContext()->module()) {
-    for (auto& block : function) {
-      for (auto& instruction : block) {
-        // Randomly decide whether to apply the transformation.
-        if (!GetFuzzerContext()->ChoosePercentage(
-                GetFuzzerContext()
-                    ->GetChanceOfReplacingAddSubMulWithCarryingExtended())) {
-          continue;
-        }
+  GetIRContext()->module()->ForEachInst([this](opt::Instruction* instruction) {
+    // Randomly decide whether to apply the transformation.
+    if (!GetFuzzerContext()->ChoosePercentage(
+            GetFuzzerContext()
+                ->GetChanceOfReplacingAddSubMulWithCarryingExtended())) {
+      return;
+    }
```

Comment:
These changes are necessary for the FuzzerPassReplaceAddsMulsWithCarryingExtended, because it turned out that all of my passes (including this one) were not included in fuzzer.cpp. Hence they were untested. I have tested this fuzzer pass and it crashed, because of broken iterator in the loop. These changes fix all the crashes that happened after I have included all of my fuzzer passes. Should I revert these changes now and open a new PR for them?

Reply:
A possible solution is to store all the necessary instructions (from a block or function etc) in the vector so that the interators are not invalidated.

Reply:
i. e. use the auxiliary space

Reply:
OK we can just leave these changes this time.

Reply:
Oh, but I don't think this particular change is a good one. I think Vasyl's suggestion is a good one.

Reply:
I have changed the pass according to this suggestion.